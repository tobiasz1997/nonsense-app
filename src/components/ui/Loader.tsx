import { FC } from 'react';
import Spinner from '@components/ui/Spinner';

type LoaderProps = {
	error?: unknown;
};

const Loader: FC<LoaderProps> = ({ error }) => {
	if (error) {
		throw error;
	}
	return (
		<div className="flex h-full w-full flex-1 items-center justify-center text-green-dark dark:text-pistachio">
			<Spinner className="h-3 fill-current" />
		</div>
	);
};

export default Loader;
