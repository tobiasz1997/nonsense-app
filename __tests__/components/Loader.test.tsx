import { render } from '@testing-library/react';
import Loader from '@components/ui/Loader';

describe('Loader', () => {
	it('should render without crashing', () => {
		const { container } = render(<Loader />);
		expect(container).toMatchSnapshot();
	});
});
