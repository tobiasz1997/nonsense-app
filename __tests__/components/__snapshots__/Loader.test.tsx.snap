// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render without crashing 1`] = `
<div>
  <div
    class="flex h-full w-full flex-1 items-center justify-center text-green-dark dark:text-pistachio"
  >
    <svg
      class="h-3 fill-current"
      height="30"
      viewBox="0 0 120 30"
      width="120"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="15"
        cy="15"
        r="15"
      >
        <animate
          attributeName="r"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="15"
          repeatCount="indefinite"
          to="15"
          values="15;9;15"
        />
        <animate
          attributeName="fill-opacity"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="1"
          repeatCount="indefinite"
          to="1"
          values="1;.5;1"
        />
      </circle>
      <circle
        cx="60"
        cy="15"
        fill-opacity="0.3"
        r="9"
      >
        <animate
          attributeName="r"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="9"
          repeatCount="indefinite"
          to="9"
          values="9;15;9"
        />
        <animate
          attributeName="fill-opacity"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="0.5"
          repeatCount="indefinite"
          to="0.5"
          values=".5;1;.5"
        />
      </circle>
      <circle
        cx="105"
        cy="15"
        r="15"
      >
        <animate
          attributeName="r"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="15"
          repeatCount="indefinite"
          to="15"
          values="15;9;15"
        />
        <animate
          attributeName="fill-opacity"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="1"
          repeatCount="indefinite"
          to="1"
          values="1;.5;1"
        />
      </circle>
    </svg>
  </div>
</div>
`;
